openmc/material.py

```diff
@@ -41,11 +41,19 @@ class Material(object):
     name : str, optional
         Name of the material. If not specified, the name will be the empty
         string.
+    temperature : str, optional
+        The temperature identifier applied to this material. The units are
+        in Kelvin and the temperature rounded to the nearest integer.
+        For example, a tempreature of 293.6K would be provided as '294K'
 
     Attributes
     ----------
     id : int
         Unique identifier for the material
+    temperature : str
```

Comment:
You're right -- with the Python API, assigning temperatures to all cells that contain the same material is easy. The other thing we could do, as a user convenience, is to have a default material temperature. So, while the ultimate temperature is still by cell and is stored as such, it would inherit from the material default temperature (which itself would inherit from a global default temperature). So you'd have:
```
default temperature
  |
  |-> material default temprature
        |
        |-> cell temperature
```

I'm plowing ahead with adding temperature dependence on the `Nuclide` and `SabData` derived types and pretty much following your outline here. I already have 1, 2, and 5 pretty much taken care of. We need to decide on what the extra input for the user would look like to decide on their temperature treatment. Right now I'm thinking they could either select to have exact temperatures (within some tolerance) or to apply temperature interpolation.